_posts/2020-01-30-end-to-end-sql.md

```diff
@@ -0,0 +1,122 @@
+In particular it marks the end of the community's year-lng effort to merge in the Blink SQL contribution from Alibaba.
+The reason the community chose to spend so much time on the contribution is that SQL works.
+It allows Flink to offer a truly unified interface over batch and streaming and makes stream processing accessible to a broad audience of developers and analysts.
+Best of all, Flink SQL is ANSI-SQL compliant, which means if you've ever used a database in the past, you already know it[^1]!
+
+While much of the work focused on improved runtime performance and greater coverage of the standard, some features focused on ease of use.
+Create table statements and catalogs are two such features.
+
+# Accessing Your Data Where It Lives
+
+Flink does not store data at rest; it is a compute engine and requires other systems to consume input from and write its output.
+Those that have used Flink's `DataStream` API in the past will be familiar with connectors that allow for interacting with external systems. 
+Flink has a vast connector ecosystem that includes all major message queues, filesystems, and databaes.
+
+<div class="alert alert-info">
+If your favorite system does not have a connector maintained in the central Apache Flink repository, check out the <a href="https://flink-packages.org">flink packages website</a>, which has a growing number of community-maintained components.
+</div>
+
+While these connectors are battle-tested and production-ready, they are written in Java and configured in code, which means they are not amenable to pure SQL or Table applications.
+Relational queries should execute against relational tables.
+
+# Create Table Statements
+
+While Flink SQL has long provided table abstractions on top some of Flink's most popular connectors, configurations were not always so straight forward.
+Begining in 1.10, Flink supports defining tables through `CREATE TABLE` statements.
+With this feature, users can now create logical tables, backed by varous external systems, in pure SQL. 
+
+This means you can define a logical table which can be read from or written to, that is abstracted away from its physical data store.
+Coupled with Flink SQL's unified approach to batch and stream processing, Flink provides a straight line from discovery to production.
+
+Users can define tables over static data sets, anything from a local CSV file to a full-fledged data lake or even Hive.
+Leveraging Flink's efficient batch processing capabilities, they can perform ad-hoc queries searching for exciting insights.
+Once something interesting is identified, buisnesses can gain real-time and continous insights by merely altering the table so that it is powered by a message queue such as Kafka.
+Because Flink guaruntees SQL queries have unified semantics over batch and streaming, users can be confident that redeploying this query as a continuous streaming application over a message queue will output identical results.
```

Comment:
```suggestion
Because Flink guarantees SQL queries have unified semantics over batch and streaming, users can be confident that redeploying this query as a continuous streaming application over a message queue will output identical results.
```

Reply:
This is why I need to stop writing blog posts in Vim, thank you :)